### DISCUSSION TITLE
Update on HelmRelease values doesn't replace pods

### DISCUSSION THREAD

**Discussion by nicogl** [fluxcd/flux2#1551] [Q&A] [answered]
I have a HelmRelease which is an overlay that overrides some configuration values of a base HelmRelease.
My HelmRelease is already deployed and running, but if I update any value in the overlay that customizes the chart values, the reconciliation doesn't seem to pick up the changes. In other words, no pods are re-created unless I remove the running HelmRelease and create it again. In fact, I get alert messages saying that Flux controllers realized about the change, but then it's not being applied, nothing changes.

Given the case I described above, shouldn't the running pods be replaced? Any idea what could be causing the issue?

**Comment by hiddeco** [comment, marked as answer]
This depends on what you changed, if you for example just changed a value in a `ConfigMap` that is part of a Helm release, you need [additional Helm specific mechanics](https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments) so that your pods also "roll".
